### PR TITLE
feat: add github issues link to regex

### LINKED ISSUE #384: Does this works with links?
**Is your feature request related to a problem? Please describe.**
I am not quite sure if this is a Feature request or a bug. but this action did not work when we had a link after the close tag 

like: close https://github.com/example/example/issues/23594

**Describe the solution you'd like**

it would be great if this is solved, or is this an issue from the action runing side?

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by a0m0rajab**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR add the extra regex to detect the issues link in the body.

## Related Tickets & Documents

Closes https://github.com/mtfoley/pr-compliance-action/issues/384

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 readme
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by a0m0rajab** [comment]
Side note: when I opened the PR felt that's really similar to open sauced, then noticed you are one of the contributors there :)

**Comment by mtfoley** [comment]
Yeah, great project 🤩

File: README.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ repo-token | `secrets.GITHUB_TOKEN` | Access token for which this action will ru
 ignore-authors | dependabot<br/>dependabot[bot] | If the action detects that the PR author is one of these logins, it will skip checks and set all outputs to `true`.
 base-comment | (see [action.yml](./action.yml)) | Preamble to any comment the action leaves on the PR.
 ignore-team-members | true | Whether to ignore in the case that the PR author is a) the owner of repository, or b) has a public membership[^1] in the org that owns the repository.
-body-regex | `(fix(es\|ed)?\|((resolve\|close)(s\|d)?)) #\d*[1-9]\d*` | Regular expression to identify whether the PR body refers to an issue[^2][^3].
+body-regex | `(fix(es|ed)?|((resolve|close)(s|d)?)) (https:\/\/github\.com\/[a-zA-Z0-9\-_]+\/[a-zA-Z0-9\-_]+\/issues\/\d+)?|(#\d*[1-9]\d*?)` | Regular expression to identify whether the PR body refers to an issue[^2][^3].
```
Comment (mtfoley) [inline README.md:69]: with the value being so long, I'd suggest having the cell content refer to action.yml as you see some of the others.

File: __tests__/checks.test.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ No issue, just passing thru.
 `
 const goodBody =
   '# Fix for lint error\n\nThis PR fixes #123\n\nNo post-deploy tasks.'
+const goodBodyWithLink =
+  '# Fix for lint error\n\nThis PR fixes https://github.com/mtfoley/pr-compliance-action/issues/384\n\nNo post-deploy tasks.'
```
Comment (mtfoley) [inline __tests__/checks.test.ts:18]: I think we'd also benefit from a test case to protect against false positives like
https://www.example.com/some/repo/123.

**Review by mtfoley — CHANGES_REQUESTED** [review]
Since you didn't change anything in main.ts or related files, I'm not surprised dist/*.js didn't change. Still, remember to run `npm run all` locally before pushing.

**Comment by a0m0rajab** [comment]
Thank you for the suggestions! Just solved it and all tests passed 👍

File: README.md (hunk 1 above)
Comment (a0m0rajab) [reply README.md]: Done!

File: __tests__/checks.test.ts (hunk 2 above)
Comment (a0m0rajab) [reply __tests__/checks.test.ts:18]: Done, thanks for that

File: dist/index.js
Comment (mtfoley) [inline dist/index.js:1]: actually really surprised that anything in dist would be affected. perhaps there's a different node version or something.  I'll probably just merge this and make sure things are cleaned up on the next release.

**Review by mtfoley — APPROVED** [review]
Thanks again for the PR @a0m0rajab!

**Comment by a0m0rajab** [comment]
Thank you for accepting it!

**Pull request merged**